pkgs/by-name/zl/zlog/package.nix

```
@@ -21,6 +21,11 @@ stdenv.mkDerivation rec {
 
   makeFlags = [ "PREFIX=${placeholder "out"}" ];
 
+  preBuild = ''
+    makeFlagsArray+=(CFLAGS="-Wno-pointer-to-int-cast -Wno-newline-eof")
+  '';
+
+
```

I'll suggest using the `NIX_CFLAGS_COMPILE` hammer. Also, is this PR actually for darwin?

```suggestion
  env.NIX_CFLAGS_COMPILE = toString (
    lib.optionals stdenv.hostPlatform.isDarwin [
      "-Wno-pointer-to-int-cast"
      "-Wno-newline-eof"
    ]
  );

```

---

i will add the optional.

but can you please explain why you would go for the hammer when the selective way of adding the flags to the build system works just fine?

---

I believe `NIX_CFLAGS_COMPILE` provides a more universal way to handle all kinds of C/C++ errors. I should have clarified that my suggestion about `NIX_CFLAGS_COMPILE` here is optional...

---

@wegank done